syncd/scripts/syncd_init_common.sh

```diff
@@ -124,74 +124,188 @@ config_syncd_cisco_8000()
     fi
 }
 
-config_syncd_bcm()
+function merge_config_bcm_files()
 {
+    to_file=$1
+    from_file=$2
+    message=$3
+    override=false
+    echo "" >> $to_file
+    echo "# Start of $message" >> $to_file
+    while read line
+    do
+        line=$( echo $line | xargs )
+        if [ ! -z "$line" ];then
+            if [ "${line::1}" == '#' ];then
+                echo $line >> $to_file
+            elif [ "$line" == "[Low Inheritance Precedence]" ];then
+                override=false
+                echo "# $line" >> $to_file
+            elif [ "$line" == "[High Inheritance Precedence]" ];then
+                override=true
+                echo "# $line" >> $to_file
+                echo "Merge properties with override $override"
+            else
+                sedline=${line%=*}
+                if grep -q $sedline $to_file ;then
+                   if $override ;then
+                      echo "Override the config $(grep $sedline $to_file) with $line in $to_file"
+                      prop=${line:0:`expr index $line =`}
+                      sed -i "/$prop/d" $to_file
+                      echo $line >> $to_file
+                   else
+                      grepline=$(grep $sedline $to_file)
+                      if [ "${grepline::1}" == '#' ];then
+                         echo $line >> $to_file
+                      else
+                         echo "Keep the config $(grep $sedline $to_file) in $to_file"
+                      fi
+                   fi
+                else
+                   echo $line >> $to_file
+                fi
+            fi
+        fi
+    done < $from_file
+    echo "# End of $message" >> $to_file
+    echo "Merged $from_file to $to_file"
+}
```

Comment:
On 7260 hwsku, syncd is not running with this change:

```
2024 Nov 25 10:04:26.043570 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd Merged /usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_b97/broadcom-sonic-th2.config.bcm to /tmp/config.bcm
2024 Nov 25 10:04:26.043962 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd Merging /etc/sai.d/config.bcm with /usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_b97/broadcom-sonic-th2.config.bcm, merge files stored in /tmp/config.bcm
2024 Nov 25 10:04:26.359627 str2-7260cx3-acs-9 INFO lldp#supervisord 2024-11-25 10:04:26,358 INFO success: rsyslogd entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2024 Nov 25 10:04:26.411278 str2-7260cx3-acs-9 DEBUG syncd#syncd: :> syncd_main: enter
2024 Nov 25 10:04:26.415024 str2-7260cx3-acs-9 NOTICE syncd#syncd: :- initialize: initializeing metadata log function
2024 Nov 25 10:04:26.415082 str2-7260cx3-acs-9 WARNING syncd#syncd: :- parseCommandLine: param -s is depreacated, use -z
2024 Nov 25 10:04:26.415106 str2-7260cx3-acs-9 WARNING syncd#syncd: :- parseCommandLine: unknown option B
2024 Nov 25 10:04:26.415128 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd /usr/bin/syncd: invalid option -- 'B'#015
2024 Nov 25 10:04:26.415150 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd Usage: syncd [-d] [-p profile] [-t type] [-u] [-S] [-U] [-C] [-s] [-z mode] [-l] [-g idx] [-x contextConfig] [-b breakConfig] [-h]#015
2024 Nov 25 10:04:26.415174 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -d --diag#015
2024 Nov 25 10:04:26.415195 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Enable diagnostic shell#015
2024 Nov 25 10:04:26.415218 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -p --profile profile#015
2024 Nov 25 10:04:26.415240 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Provide profile map file
2024 Nov 25 10:04:26.415261 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415283 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -t --startType type
2024 Nov 25 10:04:26.415348 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415373 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Specify start type (cold|warm|fast|fastfast|express)
2024 Nov 25 10:04:26.415395 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415414 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -u --useTempView
2024 Nov 25 10:04:26.415452 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415475 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Use temporary view between init and apply
2024 Nov 25 10:04:26.415532 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415556 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -S --disableExitSleep
2024 Nov 25 10:04:26.415578 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.415621 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Disable sleep when syncd crashes
2024 Nov 25 10:04:26.415658 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.419067 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -U --enableUnittests
2024 Nov 25 10:04:26.419125 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.419148 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Metadata enable unittests
2024 Nov 25 10:04:26.419169 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd #015
2024 Nov 25 10:04:26.419190 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -C --enableConsistencyCheck#015
2024 Nov 25 10:04:26.419211 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Enable consisteny check DB vs ASIC after comparison logic#015
2024 Nov 25 10:04:26.419231 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -s --syncMode#015
2024 Nov 25 10:04:26.419250 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Enable synchronous mode (depreacated, use -z)#015
2024 Nov 25 10:04:26.419274 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -z --redisCommunicationMode#015
2024 Nov 25 10:04:26.419297 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Redis communication mode (redis_async|redis_sync|zmq_sync), default: redis_async#015
2024 Nov 25 10:04:26.419319 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -l --enableBulk#015
2024 Nov 25 10:04:26.419341 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Enable SAI Bulk support#015
2024 Nov 25 10:04:26.419366 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -g --globalContext#015
2024 Nov 25 10:04:26.419389 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Global context index to load from context config file#015
2024 Nov 25 10:04:26.419410 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -x --contextConfig#015
2024 Nov 25 10:04:26.419431 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Context configuration file#015
2024 Nov 25 10:04:26.419452 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -b --breakConfig#015
2024 Nov 25 10:04:26.419474 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Comparison logic 'break before make' configuration file#015
2024 Nov 25 10:04:26.419495 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -w --watchdogWarnTimeSpan#015
2024 Nov 25 10:04:26.419518 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Watchdog time span (in microseconds) to watch for execution#015
2024 Nov 25 10:04:26.419540 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd     -h --help#015
2024 Nov 25 10:04:26.419561 str2-7260cx3-acs-9 INFO syncd#supervisord: syncd         Print out this message#015
2024 Nov 25 10:04:26.440339 str2-7260cx3-acs-9 NOTICE syncd#dsserve: child /usr/bin/syncd exited status: 256
```

Reply:
> Zhaohui,
> 
> I found the following PR which I merged from master caused the issue. [c4aea50](https://github.com/sonic-net/sonic-sairedis/commit/c4aea50ed872e1b7710d09cb0f9749bcfd8059df)
> 
> In my previous local PR for running UT, I didn't hit this issue. Recently, I merged my PR and rebase with master. The change from the following PR was merged to the syncd_init_common.sh and caused the issue.
> 
> Can you check the following PR ? Not sure how this merged to community master ? I try to remove the change of the PR change from [Stephen Sun](https://github.com/stephenxs), the issue is resolved.
> 
> Can you please confirm this again ?
> 
> Geans
> 
> 3 weeks ago
> 
> [Do not poll counters in bulk mode during initialization for objects t…](https://github.com/sonic-net/sonic-sairedis/commit/c4aea50ed872e1b7710d09cb0f9749bcfd8059df) 47 48 49 50 51 SUPPORTING_BULK_COUNTER_GROUPS=$(echo $SYNCD_VARS | jq -r '.supporting_bulk_counter_groups') if [ "$SUPPORTING_BULK_COUNTER_GROUPS" != "" ]; then CMD_ARGS+=" -B $SUPPORTING_BULK_COUNTER_GROUPS" fi

@geans-pin could you please help fix the conflicts? And  then I will test it again with your latest submit

Reply:
I don't see any conflicts. The issue is caused by the previous merged commits. Can you check with the PR owner of the previous commit by Stephen Sun? I have no idea why he add the new option -B? I don't see any UT on this PR from Stephen Sun.



Geans

Reply:
@geans-pin  To make things easier, https://github.com/sonic-net/sonic-sairedis/pull/1437 was merged before your PR, we have to rebase the code.
I will test the changes in your PR without -B option, I mean based on 202405 image, not master image.